db/mem_db.go

```diff
@@ -53,8 +53,10 @@ func (db *MemDB) DeleteSync(key []byte) {
 
 func (db *MemDB) Close() {
 	db.mtx.Lock()
-	defer db.mtx.Unlock()
-	db = nil
+	for k := range db.db {
```

Comment:
I'd argue that we need to delete anything. If you close a "DB connection", your data aren't supposed to be deleted. So, for in-memory DB, the close should be empty IMHO.

Reply:
So for other DBs, Close has different effects:
- CLevelDB: Close makes the DB unusable, so any other calls will panic https://github.com/jmhodges/levigo/blob/c42d9e0ca023e2198120196f842701bb4c55d7b9/db.go#L377-L388
- GoLevelDB: Close
https://github.com/tendermint/tmlibs/blob/9997e3a3b46db1d2f88aa9816ed0e7915dad6ac1/db/go_level_db.go#L81 --> Aborts all in-flight connections/requests and flushes all data. It advises that after close, the DB is unsafe to use


IMO an in-memory DB's close should be pretty much like flush, writing out all "unsynced" data. However, after considering your thoughts and a combination of the ideas from those DBs, I think perhaps we should instead just document that Close is a noop. If I want to actually delete all data, I can invoke db.Iterator and delete all fields. Thanks for the thoughts.

Reply:
Are you gonna update this PR?

Reply:
Yes, I am going to make it Close a noop